### DISCUSSION TITLE
Is there a way to get the derivation tree of a term in SyGuS findSynth[Next]

### DISCUSSION THREAD

**Discussion by J4K0B** [cvc5/cvc5#10238] [Q&A] [answered]
Hello, I am using the following piece of code to get a word from a given SyGuS Grammar `g`:
`cvc5::Term curr = slv.findSynth(cvc5::modes::FindSynthTarget::ENUM, g);`
Is there a way to see how the term `curr` was derived?

Consider a grammar `G` for example:
```
Start ::= S_0
S_0 ::= S_1 | S_2 | S_3 | ...
S_1 ::= 1
S_2 ::= 2 ...
```

If `curr` is `1`, I would like to get the derivation (tree) Start -> S_0 -> S_1 -> 1.

**Comment by ajreynol** [comment, marked as answer]
This kind of functionality is available for `checkSynth` as an output tag `-o sygus-sol-gterm`.  See documentation here: https://cvc5.github.io/docs/cvc5-1.0.9/output-tags.html.

In particular you will get an annotated term where each subterm involved in the derivation is labelled with the production rule that generated it, for example:
```
(sygus-sol-gterm (f (! (+ (! 1 :gterm a) (! 1 :gterm b)) :gterm a)))
```

This output tag is not available for `findSynth` at the moment, but this can easily be added, I've opened https://github.com/cvc5/cvc5/pull/10241 to do this.

**Reply by J4K0B** [reply]
Thank you so much for the quick reply!
